src/charts/helpers/style.js

```diff
@@ -38,74 +38,74 @@ module.exports = (function() {
         return defaultStyle;
     };
 
-    return {
+    /**
+     * Returns a serializer function, only run it when you know you want to serialize your chart
+     * @return {func} serializer to add styles in line to dom string
+     */
+    const initializeSerializer = () => {
+        // Mapping between tag names and css default values lookup tables. This allows to exclude default values in the result.
+        var defaultStylesByTagName = {};
+
+        // Precompute the lookup tables.
+        [].forEach.call(tagNames, (name) => {
+            if (!noStyleTags[name]) {
+                defaultStylesByTagName[name] = computeDefaultStyleByTagName(name);
+            }
+        });
 
-        /**
-         * returns serializer function, only run it when you know you want to serialize your chart
-         * @return {func} serializer to add styles in line to dom string
-         */
-        initializeSerializer() {
+        function getDefaultStyleByTagName(tagName) {
+            tagName = tagName.toUpperCase();
 
-            // Mapping between tag names and css default values lookup tables. This allows to exclude default values in the result.
-            var defaultStylesByTagName = {};
+            if (!defaultStylesByTagName[tagName]) {
+                defaultStylesByTagName[tagName] = computeDefaultStyleByTagName(tagName);
+            }
 
-            // Precompute the lookup tables.
-            [].forEach.call(tagNames, (name) => {
-                if (!noStyleTags[name]) {
-                    defaultStylesByTagName[name] = computeDefaultStyleByTagName(name);
-                }
-            });
+            return defaultStylesByTagName[tagName];
+        }
 
-            function getDefaultStyleByTagName(tagName) {
-                tagName = tagName.toUpperCase();
+        function serializeWithStyles(elem) {
+            let cssTexts = [],
+                elements,
+                computedStyle,
+                defaultStyle,
+                result;
 
-                if (!defaultStylesByTagName[tagName]) {
-                    defaultStylesByTagName[tagName] = computeDefaultStyleByTagName(tagName);
-                }
+            if (!elem || elem.nodeType !== Node.ELEMENT_NODE) {
+                // 'Error: Object passed in to serializeWithSyles not of nodeType Node.ELEMENT_NODE'
 
-                return defaultStylesByTagName[tagName];
+                return;
             }
 
-            function serializeWithStyles(elem) {
-                let cssTexts = [],
-                    elements,
-                    computedStyle,
-                    defaultStyle,
-                    result;
+            cssTexts = [];
+            elements = elem.querySelectorAll('*');
+
+            [].forEach.call(elements, (el, i) => {
```

Comment:
Why not use regular for loops or `forEach`?

Reply:
Good call. I am not sure why is this way.

I would love to refactor it a bit at some point, but for today I am just moving things around.